tests/unit/client.py

```diff
@@ -236,6 +240,21 @@ def populate_server():
                 )
 
 
+# Avoid race condition while creating, deleting networks
+def avoid_network_race_condition():
+    import sqlite3
+    global db_dir  # Value of db_dir is populated by `create_setup`.
+    db_file = db_dir+'/haas.db'
+    conn = sqlite3.connect(db_file)
+    cur = conn.cursor()
+    while True:
+        que = len(cur.execute('select * from networking_action;').fetchall())
+        if que > 0:
+            time.sleep(1)
```

Comment:
I think we should still put some upper bound on how long we're willing to wait -- maybe a minute or two. Otherwise travis will run for a *very* long time without notifying us that anything is wrong. I've had jobs just busy looping for 20 minutes before.

Doing an exponential backoff would be the "standard" approach, but I'm not picky on that point.